fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneRepartitionPlannerTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * LuceneRepartitionPlannerTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene;
+
+import com.apple.foundationdb.record.util.pair.Pair;
+import com.apple.foundationdb.tuple.Tuple;
+import com.apple.test.Tags;
+import com.google.protobuf.ByteString;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * tests for the lucene repartition planner.
+ */
+@Tag(Tags.RequiresFDB)
+public class LuceneRepartitionPlannerTest extends LuceneIndexTest {
+    @ParameterizedTest
+    @MethodSource(value = "simplePartitionConsolidationTest")
```

Comment:
In order to remove the inheritance I think you have to refer to this with a fully-qualified name:
```suggestion
    @MethodSource(value = "com.apple.foundationdb.record.lucene.LuceneIndexTest.simplePartitionConsolidationTest")
```

Although, I think it would be better to have this test have strictly more parameter sets than the integration test, but we can agree-to-disagree on that and leave as you have it, and potentially add more argument sets later.

Reply:
> In order to remove the inheritance I think you have to refer to this with a fully-qualified name:

Done. The format is `fullyQualifiedClassName#methodName`.

> Although, I think it would be better to have this test have strictly more parameter sets than the integration test, but we can agree-to-disagree on that and leave as you have it, and potentially add more argument sets later.

Ack, fair enough for now. We can update the parameter sets as needed